qiskit/opflow/gradients/circuit_gradients/lin_comb.py

```diff
@@ -665,11 +693,45 @@ def _gradient_states(
                         param_expression = gate.params[idx]
 
                         if isinstance(meas_op, OperatorBase):
-                            state = meas_op @ state
-                        elif meas_op is True:
-                            state = ListOp(
-                                [state], combo_fn=partial(self._grad_combo_fn, state_op=state_op)
-                            )
+                            state = StateFn(meas_op, is_measurement=True) @ state
+
+                        else:
+                            if aux_meas_op == Z - 1j * Y:
+                                state_z = ListOp(
+                                    [state],
+                                    combo_fn=partial(self._grad_combo_fn, state_op=state_op),
+                                )
+                                pbc = PauliBasisChange(
+                                    replacement_fn=PauliBasisChange.measurement_replacement_fn
+                                )
+                                pbc = pbc.convert(-Y ^ (I ^ (state.num_qubits - 1)))
+                                state_y = pbc[-1] @ state
+                                state_y = ListOp(
+                                    [state_y],
+                                    combo_fn=partial(self._grad_combo_fn, state_op=state_op),
+                                )
+                                state = state_z - 1j * state_y
+
+                            elif aux_meas_op == -Y:
+                                pbc = PauliBasisChange(
+                                    replacement_fn=PauliBasisChange.measurement_replacement_fn
+                                )
+                                pbc = pbc.convert(aux_meas_op ^ (I ^ (state.num_qubits - 1)))
+                                state = pbc[-1] @ state
+                                state = -1 * ListOp(
+                                    [state],
+                                    combo_fn=partial(self._grad_combo_fn, state_op=state_op),
+                                )
+                            elif aux_meas_op == Z:
+                                state = ListOp(
+                                    [state],
+                                    combo_fn=partial(self._grad_combo_fn, state_op=state_op),
+                                )
```

Comment:
Can we reduce code duplication somehow? Or at least extract content of each `if` as a function?

Reply:
Although I do see the point, this would not really reduce the lines in the code. In fact, we would only have functions with 2 lines in them.

Reply:
Ok